lib/slapi/helpers/base.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+# Client Helpers for Slapi Class
+# Its main functions are to:
+#  1. Route Messages to Client
+#     - Standard Text Chat
+#     - Emote Chat
+#     - Attachment Chat
+#  2. Set response channel based on where request type
+#     - Help request response in DM if set in Bot Config
+#     - No channel set for API requests (debug level only message)
+#     - Set to same channel as request
+#  3. Set Chat listener pre-fix base on channel
+#     - Listen for @bot, bot, or bot id as initiator if normal channel
+#     - Above plus any line that starts with a listener if direct message
+class Slapi
+  def self.chat(data = nil, channel = nil, attachment)
```

Comment:
For consistency attachment should be initialized also:

    self.chat(data = nil, channel = nil, attachment = [])

Otherwise the first two initializations are rather pointless.  More information here: http://stackoverflow.com/questions/9710303/rails-optional-argument